verilog/tools/preprocessor/verilog_preprocessor.cc

```diff
@@ -49,53 +50,169 @@ static absl::Status StripComments(const SubcommandArgsRange& args,
+    outs << u.error_message << '\n';  // for debugging.
+  //  parsing just as a trial
+  std::string post_preproc;
+  for (auto u : preprocessed_stream) post_preproc += std::string{u->text()};
+  std::string source_view{post_preproc};
+  verilog::VerilogAnalyzer analyzer(source_view, "file1", config);
+  auto analyze_status = analyzer.Analyze();
+  /* const auto& mydata = analyzer.Data().Contents(); */
+  /* outs<<mydata; */
 
-Input:
-  'file' is a Verilog or SystemVerilog source file.
-  Use '-' to read from stdin.
+  /* TODO(karimtera): regarding conditionals
+   1) Modify VerilogPreprocess config to have a configuration to generate SV
+   source codes for all possible variants. 2) Then use parser, directly from
+   VerilogAnalyzer or from VerilogParser to have less dependences. 3) Now, we
+   should have multiple trees, we need to merge them as described by Tom in
+   Verible's issue. 4) Finally, travese the tree and output the chosen path
+   based on definitions.
+  */
+  return absl::OkStatus();
+}
 
-Output: (stdout)
-  Contents of original file with // and /**/ comments removed.
-)"}},
-};
+static absl::Status GenerateVariants(const char* source_file, std::istream&,
+                                     std::ostream& outs, std::ostream&) {
+  std::string source_contents;
+  if (auto status = verible::file::GetContents(source_file, &source_contents);
+      !status.ok()) {
+    std::cerr << "ERROR: passed file can't be open\n.";
+    return status;  // check if the the file is not readable or doesn't exist.
+  }
 
-int main(int argc, char* argv[]) {
-  // Create a registry of subcommands (locally, rather than as a static global).
-  verible::SubcommandRegistry commands;
-  for (const auto& entry : kCommands) {
-    const auto status = commands.RegisterCommand(entry.first, entry.second);
-    if (!status.ok()) {
-      std::cerr << status.message() << std::endl;
-      return 2;  // fatal error
-    }
+  // Lexing the input SV source code.
+  verilog::VerilogLexer lexer(source_contents);
+  verible::TokenSequence lexed_sequence;
+  for (lexer.DoNextToken(); !lexer.GetLastToken().isEOF();
+       lexer.DoNextToken()) {
+    // For now we will store the syntax tree tokens only, ignoring all the
+    // white-space characters. however that should be stored to output the
+    // source code just like it was.
+    if (verilog::VerilogLexer::KeepSyntaxTreeTokens(lexer.GetLastToken()))
+      lexed_sequence.push_back(lexer.GetLastToken());
   }
 
-  const std::string usage = absl::StrCat("usage: ", argv[0],
-                                         " command args...\n"
-                                         "available commands:\n",
-                                         commands.ListCommands());
+  // Control flow tree constructing.
+  verilog::FlowTree control_flow_tree(lexed_sequence);
+  auto status =
+      control_flow_tree
+          .GenerateControlFlowTree();  // construct the control flow tree.
```

Comment:
It looks like the funny break in this line just happend because the comment is behind the line.
If you put it in the line before, it should be fine.
(also, this comment should probably not be there: it is just stating the obvious. Use comments for things that need description, not re-iterate what the code just says).